nmdc_server/filters.py

```diff
@@ -218,15 +218,21 @@ def join_data_object(self, query: Query) -> Query:
         return self.join_omics_processing(query)
 
 
-class EnvBroadScaleFilter(BiosampleFilter):
+class EnvoTermFilter(BiosampleFilter):
+    def join_omics_processing(self, query: Query) -> Query:
```

Comment:
Currently `join_self` has a condition in `BiosampleFilter` based on the table but `join_omics_processing` is differentiated by a subclass. For consistency could you either put a `if self.table in envo_tables` into the parent `join_omics_processing` and remove `EnvoTermFilter` or specialize the implementation of `join_self` here and remove the `if self.table in envo_tables` from `BiosampleFilter`?

Reply:
Good catch. I think that's the cleanest way to do it, and it fact, `join_omics_processing` was leveraging `join_self` prior to 688ae80.

Reply:
db5f617 removes the intermediate class that was added by a previous commit in favor of consistency with how the other filter class functions work, and consistency with the code prior to commit 688ae80